ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_indexer_grpc_utils::{
+    get_cache_coverage_status, get_cache_transactions, get_file_store_bucket_name,
+    get_redis_address,
+    storage::{generate_blob_name, TransactionsBlob, BLOB_TRANSACTION_CHUNK_SIZE},
+    CacheCoverageStatus,
+};
+use aptos_logger::{info, warn};
+use aptos_moving_average::MovingAverage;
+use aptos_protos::datastream::v1::{
+    indexer_stream_server::IndexerStream,
+    raw_datastream_response::Response as DatastreamProtoResponse, RawDatastreamRequest,
+    RawDatastreamResponse, TransactionOutput, TransactionsOutput,
+};
+use cloud_storage::Object;
+use futures::Stream;
+use redis::{Client, Commands};
+use std::{pin::Pin, thread::sleep, time::Duration};
+use tokio::sync::mpsc::{self, error::TrySendError};
+use tokio_stream::wrappers::ReceiverStream;
+use tonic::{Request, Response, Status};
+use uuid::Uuid;
+
+type ResponseStream = Pin<Box<dyn Stream<Item = Result<RawDatastreamResponse, Status>> + Send>>;
+
+pub struct DatastreamServer {
+    pub redis_client: Client,
+}
+
+impl DatastreamServer {
+    pub fn new() -> Self {
+        Self {
+            redis_client: Client::open(format!("redis://{}", get_redis_address())).unwrap(),
+        }
+    }
+}
+
+impl Default for DatastreamServer {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[tonic::async_trait]
+impl IndexerStream for DatastreamServer {
```

Comment:
we could describe what the following is doing in a few sentences, so that when we review the code, we can validate implementation matches intention?

Reply:
yes, done.

several TODOs in the stacked PRs, one follow-up PR to address them.